Simple java tasks[for review]

dmidob opened the pull request.

dmidob committed:
Removed check for null-length arrays in 1st task

fanik commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,59 @@
+public class MaxMinAverage {
+
+    public static ArrayList<Integer> generateRandomArray(int arrayLength) {
+        ArrayList<Integer> newArray = new ArrayList<Integer>();
```
You may omit "Integer" on the right side

dmidob replied:
Thanks for the comment. This was fixed.

ArtBi commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,59 @@
+        return Integer.parseInt(null);
+    }
+
+    public static int getAverageArrayValue(ArrayList<Integer> inputArray) {
```
Are you sure this method works correctly? It looks like this one just returns the second item from a given list.

dmidob replied:
Thanks for the comment. null is not parsable. So this unworkable method was removed.

ArtBi commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,59 @@
+                return averageArrayValue;
+            }
+        }
+        return Integer.parseInt(null);
```
I think it isn't a good idea to return null. 
I'm afraid You may have Null Pointer while parsing null to Integer

dmidob replied:
This was removed

ArtBi commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,59 @@
+            }
+            return maxArrayValue;
+        }
+        return Integer.parseInt(null);
```
I'm afraid You may have Null Pointer while parsing null to Integer

dmidob replied:
This was removed (both, checking array for null and prase of null as well)

ArtBi commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,59 @@
+
+    }
+
+    public static int getMaxArrayValue(ArrayList<Integer> inputArray) {
```
It would be better If you used Interface List instead of ArrayList in the method signature.

ArtBi commented on src/main/java/PrimeNumbers.java:
```diff
@@ -0,0 +1,33 @@
+public class PrimeNumbers {
+    public static ArrayList identifyPrimeNumbers() {
+        ArrayList primeNumbersList = new ArrayList();
+        for (int i = 2; i < 100; i++) {
```
Are you sure that the boundary values are right?

dmidob committed:
Updated code by removing some checks.
Removed separate "Main" Class
Added "main" methods to each executed classes.

ArtBi commented on src/main/java/ArrayEditing.java:
```diff
@@ -0,0 +1,55 @@
+                editedArray[indexForNewArray] = inputArray[i];
+                indexForNewArray++;
+            }
+            else {i++;}
```
Are you sure this "else" is needed?

fanik commented on src/main/java/ArrayEditing.java:
```diff
@@ -0,0 +1,55 @@
+                editedArray[indexForNewArray] = inputArray[i];
+                indexForNewArray++;
+            }
+            else {i++;}
```
What is the purpose of this increment?

dmidob replied:
Comment fixed - increment was identified as useless

fanik commented on src/main/java/ArrayEditing.java:
```diff
@@ -0,0 +1,55 @@
+
+
+    //This method removes element that should be removed AND THE NEXT ONE FOR SOME REASON. Investigating it.
+    private static int[] removeNumberFromArray(int[] inputArray, int numberToEdit){
```
numberToEdit - is there any other name that better represent the purpose of this variable?

dmidob replied:
variable was renamed to numberToRemove

dmidob committed:
Fixed comments from @fanik (useless increment removed, variable renamed)

dmidob committed:
Fixed comments from @ArtBi (right boundary checking conditional was amended)

fanik commented on src/main/java/BubbleSort.java:
```diff
@@ -0,0 +1,28 @@
+        for (int i = 0; i < inputArray.size(); i++) {
+            for (int j = i + 1; j < inputArray.size(); j++) {
+                if (inputArray.get(i) > inputArray.get(j)) {
+                    Collections.swap(inputArray, i, j);
```
How would you do this without the Collections swap?
Note: using collections are better in this example, however for understanding internals would be good if you can write you own equivalent of the swap function.

dmidob committed:
Added alternative swap methods for array sorting without Collections.swap usage

fanik commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,60 @@
+
+    //method for array generating and filling it by random int numbers from 0 to 10
+    static ArrayList<Integer> generateRandomArray (int arrayLength) {
+        ArrayList<Integer> newArray = new ArrayList<>();
```
I would recommend trying to work with fixed-size arrays

fanik commented on src/main/java/MaxMinAverage.java:
```diff
@@ -0,0 +1,60 @@
+         int maxArrayValue = inputArray.get(0);
+         for (Integer integer : inputArray) {
+             if (integer > maxArrayValue)
+             maxArrayValue = integer;
```
1) indentation?
2) it is better to avoid omitting brackets even for a single statement, as it may potentially grow in future and developer might miss the scope

fanik commented on src/main/java/PrimeNumbers.java:
```diff
@@ -0,0 +1,39 @@
+        for (int i = 2; i <= 100; i++) {
+            boolean isPrime = true;
+            int divider = (int) Math.sqrt(i);
+            for (int j = 2; j <= divider; j++) {
```
Can you explain why you iterating till sqrt of a current value?

fanik commented on src/main/java/BubbleSort.java:
```diff
@@ -0,0 +1,60 @@
+        for (int i = 0; i < inputArray.size(); i++) {
+            for (int j = i + 1; j < inputArray.size(); j++) {
+                if (inputArray.get(i) > inputArray.get(j)) {
+                    inputArray.set(i, inputArray.get(j) - inputArray.get(i));
```
Would be good to extract into a separate method and call swap, otherwise, it is hard to understand what it is doing unless you know the trick

dmidob committed:
1. Identations were corrected
2. Ommited bracket were added
3. Separate methods for 2 numbers swapping were added.

dmidob committed:
Package created. Classes moved to a package

dmidob committed:
All classes moved to package
New Class responsible for array generating created.
Array creating method removed from MaxMinAverage class
Unit test for getting max value added.

dmidob committed:
Added methods for int[] arrays for MaxMinAverage and Bubble Sort classes
Added unit tests for MaxMinAverage and Bubble Sort classes

dmidob committed:
Added ArrayTypesConversion class
Added unit tests for PrimeNumbers and ArrayEditing classes

ArtBi commented on src/main/java/FirstSimpleJavaTasks/ArrayEditing.java:
```diff
@@ -0,0 +1,54 @@
+package FirstSimpleJavaTasks;/*
```
You should follow Java naming convention.
